.github/workflows/ci.yml

```diff
@@ -1,67 +1,77 @@
 name: CI/CD
 
-on: [push, pull_request]
+# For now, we just manually trigger a new build when AWS Inspector identifies a security vulnerability
+# that we believe we can resolve by re-building the Docker image (because there's an apt-get update in there)
+on: [workflow_dispatch]
+
+permissions:
+  id-token: write
+  contents: read
 
 jobs:
   configure:
     name: Configure project
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
     - name: Configure
       run: ./configure.sh
   test:
     name: Test project
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
     - name: Configure
       run: ./configure.sh
     - name: Install valgrind
       run: sudo apt-get install valgrind
     - name: Test
       run: ./test.sh
+  build_and_push_staging:
+    name: Build and push to ECR (Staging)
+    needs: [configure, test]
+    if: github.ref != 'refs/heads/master'
+    environment: staging
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build and Push Docker Image
+        uses: ./.github/actions/build-and-push-docker-image
+        with:
+          aws_ecr: ${{ vars.AWS_ECR }}
+          aws_region: ${{ vars.AWS_REGION }}
+          oidc_role: ${{ vars.AWS_GITHUB_OIDC_ROLE }}
   performance:
     name: Performance tests
     needs: [configure, test]
     if: github.ref == 'refs/heads/master'
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-    - name: Configure
-      run: ./configure.sh
-    - name: Install valgrind
-      run: sudo apt-get install valgrind
-    - name: Test
-      run: ./performance.sh
-  docker:
-    name: Build and push to Dockerhub
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Configure
+        run: ./configure.sh
+      - name: Install valgrind
+        run: sudo apt-get install valgrind
```

Comment:
nice! should we use this elsewhere?

Reply:
Perhaps?  It's coming directly out of the underlying repo I forked this off of so I kept it